src/main/java/com/nike/cerberus/domain/environment/StackName.java

```diff
@@ -16,47 +16,94 @@
 
 package com.nike.cerberus.domain.environment;
 
+import com.google.common.collect.ImmutableList;
+import com.nike.cerberus.ConfigConstants;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+
 /**
  * Describes the stacks that make up Cerberus.
  */
-public enum StackName {
-    BASE("base"),
-    VAULT("vault"),
-    CMS("cms"),
-    GATEWAY("gateway"),
-    VPC("vpc"),
-    DATABASE("database"),
-    SECURITY_GROUPS("security-groups"),
-    LOAD_BALANCER("load-balancer"),
-    ROUTE53("route53"),
-    WAF("web-app-firewall");
+public class StackName {
+
+    public static final StackName BASE = new StackName("base", "base.yaml");
+    public static final StackName VAULT = new StackName("vault", null);
+    public static final StackName CMS = new StackName("cms", "cms-cluster.yaml");
+    public static final StackName GATEWAY = new StackName("gateway", null);
+    public static final StackName VPC = new StackName("vpc", "vpc.yaml");
+    public static final StackName DATABASE = new StackName("database", "database.yaml");
+    public static final StackName SECURITY_GROUPS = new StackName("security-groups", "security-groups.yaml");
+    public static final StackName LOAD_BALANCER = new StackName("load-balancer", "load-balancer.yaml");
+    public static final StackName ROUTE53 = new StackName("route53", "route53.yaml");
+    public static final StackName WAF = new StackName("web-app-firewall", "web-app-firewall.yaml");
+
+    public static final ImmutableList<StackName> ALL_STACKS = ImmutableList.of(BASE, VAULT, CMS, GATEWAY, VPC, DATABASE, SECURITY_GROUPS, LOAD_BALANCER, ROUTE53, WAF);
+
+    private static final String TEMPLATE_PATH_ROOT = "/cloudformation/";
+
+    private final Logger logger = LoggerFactory.getLogger(StackName.class);
+
 
     private final String name;
+    private final String templatePath;
 
-    StackName(final String name) {
+    private StackName(final String name, final String yamlFileName) {
```

Comment:
nit: we should change `yamlFileName` to `cloudFormationFileName` or `cfFileName`